ui/asserts.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+/**
+ * 1. [ASSERTS_ENABLED] env enables or disables it on compile-time.
+ * 2. [window.app.pause_asserts] is the runtime control (e.g. to temporarily disable it).
+ *
+ * @param condition
+ * @param msg
+ */
+export function assert(condition: any, msg: string = 'Assertion failed') {
+  // @if ASSERTS_ENABLED='true'
```

Comment:
Do we want to ensure that production deployment has this set one way or another before merging, or also check if production?

Reply:
it should default to fault if not set right?

Reply:
I pre-empted and added `ASSERTS_ENABLED=false` in the production instance, but not sure how many instances we have?

I went with "defaulting to true" because:
- If it was false, contributors/developers will never enable it unless specifically told to so.  We want errors to surface early.
- Can't auto disable it when `env.production=true` because it should also be enabled on test instances, since that's where we want to capture bugs too.
- If "automatically enabled when `env.production=false` or if `ASSERTS_ENABLED=true`", this seems confusing semantically because if I'm in localhost and sets `ASSERTS_ENABLED=false`, it will still be enabled.
- Tried to do some URL checking, but our defaults are odysee.com.  But again, `ASSERTS_ENABLED=false` will have no meaning if we do these automated stuff.

----

- [ ] On the fence about this too.  Will think about it more to prioritize avoiding it in real odysee.com. Most likely will just ignore the semantic problem ...